internal/worker/api/server_test.go

```diff
@@ -0,0 +1,559 @@
+package api
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/renatoaguimaraes/job-scheduler/internal/worker/proto"
+	"github.com/renatoaguimaraes/job-scheduler/pkg/worker/conf"
+	"github.com/stretchr/testify/assert"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/status"
+)
+
+var config = conf.Config{ServerAddress: "localhost:8080", LogFolder: os.TempDir()}
+
+func TestStartAuthnAuthzAdminUser(t *testing.T) {
+	// load server credentials
```

Comment:
nit: all of this server setup code can be extracted into a helper func and reused in other tests